src/prefect/server/database/configurations.py

```diff
@@ -256,6 +262,20 @@ async def engine(self) -> AsyncEngine:
                     application_name=self.connection_app_name
                 )
 
+            if PREFECT_SERVER_DATABASE_SQLALCHEMY_CONNECT_ARGS_TLS_ENABLED.value():
+                pg_ctx = ssl.create_default_context(
+                    purpose=ssl.Purpose.SERVER_AUTH,
+                    cafile=PREFECT_SERVER_DATABASE_SQLALCHEMY_CONNECT_ARGS_TLS_CA_FILE.value(),
+                )
+                pg_ctx.minimum_version = ssl.TLSVersion.TLSv1_2
+                pg_ctx.load_cert_chain(
+                    certfile=PREFECT_SERVER_DATABASE_SQLALCHEMY_CONNECT_ARGS_TLS_CERT_FILE.value(),
+                    keyfile=PREFECT_SERVER_DATABASE_SQLALCHEMY_CONNECT_ARGS_TLS_KEY_FILE.value(),
+                )
+                pg_ctx.check_hostname = PREFECT_SERVER_DATABASE_SQLALCHEMY_CONNECT_ARGS_TLS_CHECK_HOSTNAME.value()
+                pg_ctx.verify_mode = ssl.CERT_REQUIRED
+                connect_args["ssl"] = pg_ctx
+
```

Comment:
we should use `get_current_settings().server.database.sqlalchemy.foo` access here

Reply:
got it, I will take a look to see how to implement it, please ignore my conitnuous commits, I will rebase it later on.

Reply:
I just rebase latest main branch to fix the compability test, but I still run into issue for the `Integration tests`

Reply:
@zzstoatzz  rebased with latest fix, please help to check